lambda/getSimDetailsFromCache.ts

```diff
@@ -0,0 +1,89 @@
+import { type DynamoDBClient, QueryCommand } from '@aws-sdk/client-dynamodb'
+import { ErrorType, type ErrorInfo } from './queueJob.js'
+import { unmarshall } from '@aws-sdk/util-dynamodb'
+
+export type SimDetails = {
+	usedBytes: number
+	totalBytes: number
+}
+
+export const getSimDetailsFromCache =
+	(db: DynamoDBClient, cacheTableName: string) =>
+	async (
+		iccid: string,
+	): Promise<
+		| { error: ErrorInfo }
+		| { success: { timestamp: Date; simDetails: SimDetails } }
+	> => {
+		try {
+			const simDetails = db.send(
+				new QueryCommand({
+					TableName: cacheTableName,
+					KeyConditionExpression: 'iccid = :iccid',
+					ExpressionAttributeValues: {
+						[':iccid']: {
+							S: String(iccid),
+						},
+					},
+					ProjectionExpression: 'usedBytes, totalBytes, SIMExisting,ts',
+					ScanIndexForward: false,
+					Limit: 1,
+				}),
+			)
+			const items = (await simDetails).Items?.[0]
+
+			//No information about SIM in cache
+			if (items === undefined) {
+				throw new Error('NOT_FOUND')
+			}
+
+			//SIM is not existing
+			const SIMexisting = unmarshall(items).SIMExisting as boolean
+			if (!SIMexisting) {
+				throw new Error('NOT_EXISTING')
+			}
+
+			const responseObject: { timestamp: Date; simDetails: SimDetails } = {
+				timestamp: new Date(unmarshall(items).ts),
+				simDetails: {
+					usedBytes: unmarshall(items).usedBytes,
+					totalBytes: unmarshall(items).totalBytes,
+				},
+			}
+			return {
+				success: responseObject,
+			}
+		} catch (err) {
```

Comment:
Simplify this by removing the try/catch and returning the respective Error instance where it happens.